ai-data/generative-apis/how-to/use-structured-outputs.mdx

```diff
@@ -0,0 +1,242 @@
+tags: chat-completitions-api
+dates:
+  validation: 2024-09-17
+  posted: 2024-09-17
+---
+
+
+Structured outputs allow users to get consistent, machine-readable (JSON format) responses from language models (LLMs).
+By specifying a response format when using the [Chat Completions API](/ai-data/generative-apis/api-cli/using-chat-api/), you can ensure that responses are returned in a predefined JSON structure.
+There are two main modes for generating structured outputs: **Object Mode** (schemaless) and **Schema Mode** (deterministic, structured output).
+
+You can interact with text models using several ways:
+- The Scaleway [console](https://console.scaleway.com) will soon provide a complete [playground](/ai-data/generative-apis/how-to/query-text-models/#accessing-the-playground), aiming to test models, adapt parameters, and observe how these changes affect the output in real-time.
+- Via the [Chat API](/ai-data/generative-apis/how-to/query-text-models/#querying-text-models-via-api)
+
+<Macro id="requirements" />
+
+- Access to this service is restricted while in beta. You can request access to the product by filling out a form on the Scaleway's [betas page](https://www.scaleway.com/en/betas/#generative-api).
+- A Scaleway account logged into the [console](https://console.scaleway.com)
+- [Owner](/identity-and-access-management/iam/concepts/#owner) status or [IAM permissions](/identity-and-access-management/iam/concepts/#permission) allowing you to perform actions in the intended Organization
+- A valid [API key](/identity-and-access-management/iam/how-to/create-api-keys/) for API authentication
+- Python 3.7+ installed on your system
+
+## Types of structured outputs
+
+- **JSON mode** (schemaless):
+   - Type: `{"type": "json_object"}`
+   - This mode is non-deterministic and allows the model to output a JSON object without strict validation.
+   - Useful for flexible outputs when you expect the model to infer a reasonable structure based on your prompt.
+   - JSON Mode is older and has been used by developers since early API implementations.
+
+- **Structured outputs (schema mode)** (deterministic/structured):
+   - Type `{"type": "json_schema"}`
+   - This mode enforces a strict schema format, where the output adheres to the predefined structure.
+   - Supports complex types and validation mechanisms as per the JSON schema specification (e.g., `anyOf`, `oneOf`, `allOf`).
+   - Structured outputs is newer and has been implemented by OpenAI in September 2023 to enable stricter, schema-based response formatting.
+
+<Message type="note">
+    - All LLMs on the Scaleway library support **JSON mode** and **Structured outputs**, however, quality of results will vary in the schemaless JSON mode.
+    - JSON mode: It is important to explicitly ask the model to generate JSON output either in system prompt or user prompt. To prevent infinite generations, you are encouraged to ask the model for short JSON objects.
```

Comment:
```suggestion
    - JSON mode: It is important to explicitly ask the model to generate a JSON output either in system prompt or user prompt. To prevent infinite generations, you are encouraged to ask the model for short JSON objects.
```